bridge/src/worker.rs

```diff
@@ -0,0 +1,57 @@
+use crate::connection::Client;
+use crate::reactor::{Reactor, ReactorError, Request, Response};
+use prost::Message;
+use thiserror::Error;
+use temporal_sdk_core_api::worker::{WorkerConfigBuilder, WorkerConfigBuilderError};
+use tokio::runtime::{Runtime};
+use std::sync::Arc;
+
+#[derive(Error, Debug)]
+pub enum WorkerError {
+    #[error(transparent)]
+    EncodeError(#[from] prost::EncodeError),
+
+    #[error(transparent)]
+    InvalidWorkerOptions(#[from] WorkerConfigBuilderError),
+
+    #[error(transparent)]
+    ReactorError(#[from] ReactorError),
+}
+
+pub struct Worker {
+    worker: Arc<temporal_sdk_core::Worker>,
+    reactor: Arc<Reactor>,
+}
+
+impl Worker {
+    // TODO: Extend this to include full worker config
+    pub fn create(runtime: Arc<Runtime>, client: &Client, namespace: &str, task_queue: &str) -> Result<Worker, WorkerError> {
+        let reactor = Reactor::new(runtime);
+        let config = WorkerConfigBuilder::default()
+            .namespace(namespace)
+            .task_queue(task_queue)
+            .build()?;
+
+        let request = Request::InitWorker { config, client: client.clone() };
+        let response = reactor.process(request)?;
+
+        if let Response::Worker { worker } = response {
+            return Ok(Worker { worker: Arc::new(worker), reactor: Arc::new(reactor) })
+        } else {
+            panic!("Unexpected response type from reactor. Expected Response::Worker");
+        }
```

Comment:
That makes sense, yes 👍

Reply:
This doesn't work — the compiler complains that you haven't check for all the possible `response` types

Reply:
What Chad wrote won't quite work - but, this function already returns a result. I would just return an error here if there's a type mismatch rather than panicking